src/gui/qgsowssourceselect.cpp

```diff
@@ -114,10 +112,36 @@ QgsOWSSourceSelect::QgsOWSSourceSelect( const QString &service, QWidget *parent,
     mCacheWidget->hide();
   }
 
+  prepareExtent();
+
   // set up the WMS connections we already know about
   populateConnectionList();
 }
 
+QgsMapCanvas *QgsOWSSourceSelect::mapCanvas()
+{
+  return mMapCanvas;
+}
```

Comment:
This one can be removed -- if you add the call to QgsAbstractDataSourceWidget::setMapCanvas in setMapCanvas, then the base class mMapCavnas member will be set and it will be fine to just call the base class "mapCanvas()" implementation to retrieve this widget.